src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy

```diff
@@ -1476,159 +1476,168 @@ class SaagieClient {
+						throw new GradleException("Couldn't get id for the job after creation or update")
+					}
+					if (parsedNewlyCreatedJob?.id) {
+						jobToImport.id = parsedNewlyCreatedJob?.id
+					} else {
+						jobToImport.id = foundNameId
+					}
+					versions.each {
+						JobVersion jobVersionFromVersions = ImportJobService.convertFromMapToJsonVersion(it)
+						addJobVersion(jobToImport, jobVersionFromVersions)
+					}
+					
+					response.job.last() << [
+							versions : versions.size()
+					]
 				}
 				
-				def newlyCreatedPipeline = createProjectPipeline(pipelineToImport, pipelineVersionToImport)
-				parsedNewlyCreatedPipeline = slurper.parseText(newlyCreatedPipeline)
 			}
 			
-			response.pipeline << [
-					id   : pipeline.key,
-					name : newMappedPipeline.pipeline.name
-			]
-			
-			if (versions && versions.size() >= 1) {
-				versions.each {
-					if (!parsedNewlyCreatedPipeline?.id && !pipelineFoundId) {
-						throw new GradleException("Couldn't get id for the pipeline after creation or update")
+			def listPipelines = null
+			def callbackPipelinesToDebug = { newMappedPipeline, pipeline, id, versions, newlistJobs ->
+				listPipelines = getPipelineListByNameAndId()
+				def pipelineToImport = newMappedPipeline.pipeline
+				def pipelineVersionToImport = newMappedPipeline.pipelineVersion
+				boolean nameExist = false
+				def pipelineFoundId = null
+				if (listPipelines) {
+					listPipelines.each {
+						if (it.name == newMappedPipeline.pipeline.name) {
+							pipelineFoundId = it.id
+							nameExist = true
+						}
 					}
-					if (parsedNewlyCreatedPipeline?.id) {
-						pipelineToImport.id = parsedNewlyCreatedPipeline?.id
-					} else {
-						pipelineToImport.id = pipelineFoundId
+					// change the job to Queue so we can remove the first
+				}
+				
+				pipelineToImport.id = pipelineFoundId
+				def parsedNewlyCreatedPipeline = null
+				
+				if (listPipelines && nameExist) {
+					updatePipelineVersion(pipelineToImport, pipelineVersionToImport)
+				} else {
+					if (versions && versions.size() >= 1) {
+						versions = versions as Queue
+						def firstVersionInV1Format = versions.poll()
+						PipelineVersion firstVersion = ImportPipelineService.convertFromMapToJsonVersion(firstVersionInV1Format, newlistJobs)
+						pipelineVersionToImport = firstVersion
 					}
-					PipelineVersion pipelineVersionFromVersions = ImportPipelineService.convertFromMapToJsonVersion(it, newlistJobs)
-					updatePipelineVersion(pipelineToImport, pipelineVersionFromVersions)
 					
+					def newlyCreatedPipeline = createProjectPipeline(pipelineToImport, pipelineVersionToImport)
+					parsedNewlyCreatedPipeline = slurper.parseText(newlyCreatedPipeline)
 				}
 				
-				response.pipeline.last() << [
-						versions: versions.size()
+				response.pipeline << [
+						id   : pipeline.key,
+						name : newMappedPipeline.pipeline.name
 				]
+				
+				if (versions && versions.size() >= 1) {
```

Comment:
`if (versions?.size() >= 1) {`

But I'm not sure for this one, check before if `?` operator it's working on methods